homeworks/andrii.tkachenko_Hodr1337/2-logic/logic.js

```diff
@@ -1,28 +1,51 @@
-const firstNumber = 20;
-const secondNumber = 22;
-const operation = '*';
-function result(params) {
-    return console.log(params);
-}
-switch (operation) {
-    case '*':
-        result(firstNumber * secondNumber);
-        break;
-    case '/':
-        result(firstNumber / secondNumber);
-        break;
-    case '+':
-        result(firstNumber + secondNumber);
-        break;
-    case '-':
-        result(firstNumber - secondNumber);
-        break;
-    case '**':
-        result(firstNumber ** secondNumber);
-        break;
-    case '%':
-        result(firstNumber % secondNumber);
-        break;
-    default:
-        result('Something went wrong');
+const readline = require('readline');
+
+const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+});
+
+function calculactionlogic(a, b, opss) {
+    let calculationResult;
+    if (opss === '+') {
+        calculationResult = a + b;
+    } else if (opss === '-') {
+        calculationResult = a - b;
+    } else if (opss === '*') {
+        calculationResult = a * b;
+    } else if (opss === '/') {
+        calculationResult = a / b;
+    } else if (opss === '**') {
+        calculationResult = a ** b;
+    } else if (opss === '%') {
+        calculationResult = a % b;
+    } else {
+        console.log('You entered wrong operator. Try again please');
+    }
+    return calculationResult;
 }
+
+rl.question('Enter first number ', (answer) => {
+    const firstNumber = Number(answer);
+    // eslint-disable-next-line no-restricted-globals
+    if (isNaN(firstNumber)) {
+        console.log('You entered not a number. Try again please');
+        return rl.close();
+    }
+
+    rl.question('Enter operator ', (operator) => {
+        rl.question('Enter second number ', (answer2) => {
+            const secondNumber = Number(answer2);
+            // eslint-disable-next-line no-restricted-globals
+            if (isNaN(secondNumber)) {
```

Comment:
`eslint-disable-next-line no-restricted-globals` is intended to block invalid globals you should not use. In this case, it's preferable to use `Number.isNaN` instead of a regular `NaN`.

> isNaN() method returns true if a value is Not-a-Number. Number. isNaN() returns true if a number is Not-a-Number.

So please, fix and review all the code parts where you use a regular `NaN`.